shared/nav.ios.js

```diff
@@ -152,11 +151,12 @@ class Nav extends Component {
       return this._renderContent(activeTab, module)
     }
 
+    const enablePushPrompt = this.props.provisioned && this.props.permissionsPrompt
     const tabContent = mapValues(tabs, ({module}, tab) => (activeTab === tab && this._renderContent(tab, module)))
-
     return (
       <Box style={{flex: 1}}>
-        <TabBar onTabClick={this.props.switchTab} selectedTab={activeTab} username={this.props.username} badgeNumbers={{[folderTab]: this.props.folderBadge}} tabContent={tabContent} />
+        {!enablePushPrompt && <TabBar onTabClick={this.props.switchTab} selectedTab={activeTab} username={this.props.username} badgeNumbers={{[folderTab]: this.props.folderBadge}} tabContent={tabContent} />}
+        <Push prompt={enablePushPrompt} />
```

Comment:
No matter if prompt is enabled, I want the Push component to be mounted since that is where it is configured (see Push#componentDidMount -> configurePush).

I also thought that it's nice to have all the push stuff in a single component, even if there is no UI being displayed (if prompt disabled).